### PR TITLE
[fuzzer] Create a fuzzer for Transation Universe

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

We've implemented an AccountUniverse previously that can:
1. Generate a random sets of accounts
2. Generate a random sets of transactions and execute them to see if they matches expectation, including:
    - p2p payment txn
    - mint txn
    - key rotation txn
    - invalid txn 

This PR refactors the existing code so that fuzzer can use this existing framework to generate transactions that might break the system.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#[derive(Clone, Debug, Default)]
+pub struct LanguageTransactionExecution;
+
+impl FuzzTargetImpl for LanguageTransactionExecution {
+    fn name(&self) -> &'static str {
+        module_name!()
+    }
+
+    fn description(&self) -> &'static str {
+        "Storage save blocks"
```
Comment (sunshowers) [inline testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:25]: This needs to be updated.

File: testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    fn fuzz(&self, data: &[u8]) {
+        let passthrough_rng =
+            test_runner::TestRng::from_seed(test_runner::RngAlgorithm::PassThrough, &data);
+
+        let config = test_runner::Config::default();
+        let mut runner = TestRunner::new_with_rng(config, passthrough_rng);
+
+        let txn_trategy = vec(all_transactions_strategy(0, 1_000_000), 1..40);
```
Comment (sunshowers) [inline testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:42]: `strategy`

File: testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        let txn_trategy_tree = match txn_trategy.new_tree(&mut runner) {
```
Comment (sunshowers) [inline testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:43]: `strategy_tree`

File: testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        };
+        let txn = txn_trategy_tree.current();
+
+        let universe_strategy =
+            AccountUniverseGen::strategy(2..20, log_balance_strategy(10_000_000));
+        let universe_strategy_tree = match universe_strategy.new_tree(&mut runner) {
+            Ok(x) => x,
+            Err(_) => return,
+        };
+        let universe = universe_strategy_tree.current();
```
Comment (sunshowers) [inline testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:55]: Could you use `ValueGenerator` for these?

File: testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+        let config = test_runner::Config::default();
+        let mut runner = TestRunner::new_with_rng(config, passthrough_rng);
+
+        let txn_trategy = vec(all_transactions_strategy(0, 1_000_000), 1..40);
+        let txn_trategy_tree = match txn_trategy.new_tree(&mut runner) {
+            Ok(x) => x,
+            Err(_) => return,
+        };
+        let txn = txn_trategy_tree.current();
```
Comment (sunshowers) [inline testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:47]: I'd call this `txns` since it's a vector.

**Comment by sunshowers** [comment]
@bors-libra delegate=runtian-zhou

This looks fantastic! Thank you for this incredible work. This also gives us some useful examples for automatically converting proptests to fuzzers.

File: testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs (hunk 4 above)
Comment (runtian-zhou) [reply testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:55]: ValueGenerator doesn't have a method to initialize with a specific RNG. Should I add this method to `ValueGenerator`?

Comment (sunshowers) [reply testsuite/libra-fuzzer/src/fuzz_targets/language_transaction_execution.rs:55]: yeah!

**Pull request closed without merging**